analyzers/tests/SonarAnalyzer.UnitTest/Helpers/TestAnalysisComparerTest.cs

```diff
@@ -0,0 +1,74 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2021 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Generic;
+using System.Linq;
+using FluentAssertions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarAnalyzer.Common;
+using SonarAnalyzer.Rules.CSharp;
+using SonarAnalyzer.UnitTest.TestFramework;
+
+namespace SonarAnalyzer.UnitTest.Helpers
+{
+    [TestClass]
+    public class TestAnalysisComparerTest
+    {
+        [TestMethod]
+        public void CompareShouldListAllDifferences()
+        {
+            const string expected = @"Comparing actual issues with expected:
+Language version: C# 10.
+
+TestAnalysisComparer_1.cs
+Line 8: unexpected primary issue S3168 with message 'Return 'Task' instead.'.
```

Comment:
Can be removed for simplicity
```suggestion
Line 8: unexpected primary issue S3168 'Return 'Task' instead.'.
```